lib/barbeque/message/job_execution.rb

```diff
@@ -7,6 +7,19 @@ class JobExecution < Base
       attr_reader :application # [String] To specify `job_definitions.application_id`
       attr_reader :job         # [String] To specify `job_definitions.name`
 
+      # Creates JobExecution messages from a notification which came from subscribed SNS topic
+      def self.create_message_from_sns_notification(raw_message, message_body, job_queue:)
+        topic_arn = message_body['TopicArn']
+        subscription = SNSSubscription.find_by!(topic_arn: topic_arn, job_queue_id: job_queue.id)
+        body = {
+          'Type' => 'JobExecution',
+          'Application' => subscription.job_definition.app.name,
```

Comment:
I believe you mean `has_one :app, through: job_definition`. done it. 253910bb4f360d1eda7e89cee58f1fc60302b165